### ISSUE TITLE
[uikit-workshop] View All for sub types is broken

### ISSUE THREAD

**Issue by engelfrost** [pattern-lab/patternlab-node#890]
I am using Pattern Lab Node `v3.0.0-alpha.16` on `Mac`, with Node `v9.6.1`, using a `Vanilla` Edition.

![image](https://user-images.githubusercontent.com/11315/42571796-4a69e944-8518-11e8-8723-f6fb778489a3.png)

This sound similar to #835. 

##### Expected Behavior

Load the view localhost:3000/?p=viewall-templates-onboarding (which works if I type the URL)

![image](https://user-images.githubusercontent.com/11315/42571897-9b2984fc-8518-11e8-8c8d-764e0fc9d090.png)

##### Actual Behavior

![image](https://user-images.githubusercontent.com/11315/42571878-8df3feb6-8518-11e8-981b-8c656b3e2406.png)

##### Steps to Reproduce

Try to open a View all page for a sub type.

**Comment by Gumbtastic** [comment]
I'm not sure, if this is already known, but this bug only occours, when the patterns are orderd by double digit prefix like so: "00-global". Having the patterns orderd alphabetically causes no problems.

**Comment by jeffwhitfield** [comment]
This is still an issue with v3.0 beta 1. Just installed a fresh copy using the Mustache demo. "View All" links just don't work. For instance, going to Organisms->Global->View All results in the behavior described above. As mentioned above, doing away with ordered groups seems to work...but considering most people want to order their patterns, this definitely needs to be addressed.

**Comment by latviancoder** [comment]
Experiencing the same problem with v3.0.0-beta.1.

**Comment by BlushingFox** [comment]
Just installed v3.0.0-beta.1 and having the same issue.

**Comment by BlushingFox** [comment]
> I'm not sure, if this is already known, but this bug only occours, when the patterns are orderd by double digit prefix like so: "00-global". Having the patterns orderd alphabetically causes no problems.

Just installed v3.0.0-beta.1 and having the same issue.

**Comment by bmuenzenmeyer** [comment]
@sghoweri I've narrowed down the difference in each approach to `ui_builder.js` `injectDocumentationBlock()`. The broken subtypes already have a `docPattern` found (due to their having markdown I suspect). 

I can monkey-patch the correct `patternLink` a number of different ways here, but the main problem to me seems like [loadPattern](https://github.com/pattern-lab/patternlab-node/blob/dev/packages/core/src/lib/loadPattern.js#L71) assumes this property change will update the "computed" property [`patternLink`](https://github.com/pattern-lab/patternlab-node/blob/dev/packages/core/src/lib/object_factory.js#L77)

So, I want to fix that. But even monkey-patching this quick still gives me this error:

![image](https://user-images.githubusercontent.com/298435/48003218-71658500-e0d3-11e8-90a9-253ce8dede2e.png)


Likely there is an additional bug underneath this one. My next steps will be to better understand the state of patterns within `injectDocumentationBlock()`, since they are returned as found (broken upstream) or created just-in-time (working).

Stay tuned

**Comment by james-nash** [comment]
We've recently migrated from PL2 to PL3 and found the same issue. Glad to see it's already being worked on.

In case it's of any use, we have a hosted copy of our pattern library here: http://style-staging.buildit.digital/ where you can easily reproduce the issue - for example, try "Molecules" > "Lists" > "View all". The corresponding source code can be found here: https://github.com/buildit/gravity-ui-sass/tree/develop/src/styleguide

FWIW, most of our patterns have markdown files so perhaps that adds some weight to your suspicions (and those raised in #970) that this triggers the bug somehow.

Hope that helps!

**Comment by bmuenzenmeyer** [comment]
Hoping to devote an hour or so to this soon!

**Comment by sghoweri** [comment]
@bmuenzenmeyer I think I might have a fix for this -- PR should be up shortly!

**Comment by willthemoor** [comment]
Just having an initial play with 3.0 and ran into this issue. I'm using `edition-twig` and `starterkit-twig-demo` without any customizations or changes from what npm installs. 

Not sure if this report is helpful at this point but sharing just in case. 

**tl;dr**: markdown files didn't seem to matter. Numeric folder naming did.

After reading through issues I tried dumping all markdown files by running this from the `_patterns` folder. 
```sh
find . -name '*.md' -print0 | xargs -0 rm
```

but the View All issue persisted.

Then, I went into the `03-organisms` folder and removed all numerical prefixes from the folder names. 

```
/00-global/   --> /global/
/01-article/  --> /article/
/02-comments/ --> /comments/
/03-sections/ --> /sections/
```

Restarted the PL server and View All worked as expected.
